### ISSUE TITLE
Rename `tmpdir()` to `mktmp()`.

### ISSUE THREAD

**Issue by ariporad** [shelljs/shelljs#305]
To comply with standard unix naming.

@nfischer: Thoughts?

**Comment by nfischer** [comment]
So would we deprecate the old name, or just get rid of it?

Also, I believe `shell.tempdir()` only returns the name of the temporary directory, it doesn't make it

**Comment by TimothyGu** [comment]
:+1:

**Comment by aurium** [comment]
Folks, the main feature of `mktemp` is to create a temp file. A directory is created if you ask it explicitly with a `-d`:
```bash
$ mktemp
/tmp/tmp.626bNb2p7O
$ ls -l /tmp/tmp.626bNb2p7O
-rw------- 1 aurium users 0 Jan 25 18:43 /tmp/tmp.626bNb2p7O
$ file /tmp/tmp.626bNb2p7O
/tmp/tmp.626bNb2p7O: empty

$ mktemp -d
/tmp/tmp.3AtqZ2QIbT
$ ls -l /tmp/tmp.3AtqZ2QIbT
total 0
$ file /tmp/tmp.3AtqZ2QIbT
/tmp/tmp.3AtqZ2QIbT: directory
```

Rename `tmpdir()` to `mktmp()` will not be a good Unix mimic. _I think..._

**Comment by ariporad** [comment]
Well, we could add creation functionality.

**Comment by nfischer** [comment]
I'm not sure how often it'd be used to justify pulling it into shelljs instead of relying on exec(). If cross-platform is necessary, you can always have an if statement that has the exec for UNIX mktmp in one branch and Window's equivalent in the other

**Comment by aurium** [comment]
IMHO, I believe its a very important feature because it shorthand a very important good practices.

Most of my bash scripts and from my work friends must create temp files and it should be done by some tool (or function) that can do it by the right way.

If shelljs do not provide a cross-platform mktmp, some users may follow the easy path writing on pwd within an untested filename. So... shelljs can promote a good practices making it easy to create temp files or directories.

**Comment by nfischer** [comment]
I think tempdir() exists for this purpose (and solves it well enough for me). I'll gladly look over any PRs for this, though

**Comment by aurium** [comment]
Yes! For sure. `tempdir` will not make temp files, however temp directories are way more frequent than temp files, and will work when we need for one file only.

I was worried because the issue proposal is to rename `tmpdir()` to `mktmp()`, without talking about the `-d`.

`tempdir` can do the job, still i support this proposal because it makes Shelljs to use the same Unix nomenclature. :-)

**Comment by nfischer** [comment]
Update:

 - #488 has an initial implementation of this.
 - POSIX does not specify `mktemp` last I checked, so the Mac OS and Linux versions differ quite a bit. I have reservations about this disparity, which means I'm leaning more toward implementing this as a [ShellJS plugin](https://github.com/shelljs/shelljs/wiki/Using-ShellJS-Plugins). This way, anyone can implement it.
 - If this does get implemented as part of ShellJS, we could have an option to automatically clean up temp files when the script exits (the default behavior would be consistent with unix).
